legal-api/src/legal_api/resources/v2/business/business.py

```diff
@@ -167,19 +167,24 @@ def search_businesses():
         draft_query = db.session.query(Filing).filter(
             and_(Filing.temp_reg.in_(identifiers), Filing.business_id.is_(None)))
 
-        # parse results
         bus_results = [x.json(slim=True) for x in bus_query.all()]
-        draft_results = [
-            {
-                'identifier': x.temp_reg,
-                'legalType': x.json_legal_type,
-                **({'nrNumber': x.json_nr} if x.json_nr else {}),
-                'legalName': (x.filing_json.get('filing', {})
-                              .get(x.filing_type, {})
-                              .get('nameRequest', {})
-                              .get('legalName')),
-                'draftType': Filing.FILINGS.get(x.filing_type, {}).get('temporaryCorpTypeCode')
-            } for x in draft_query.all()]
+
+        draft_results = []
+        for draft_dao in draft_query.all():
+            draft = {
+                'identifier': draft_dao.temp_reg,  
+                'legalType': draft_dao.json_legal_type
+            }
+            if draft_dao.json_nr:
+                draft['nrNumber'] = draft_dao.json_nr
+            draft['legalName'] = (draft_dao.filing_json.get('filing', {})
+                                        .get(draft_dao.filing_type, {})
+                                        .get('nameRequest', {})
+                                        .get('legalName'))
+            draft['draftType'] = Filing.FILINGS.get(draft_dao.filing_type, {}).get('temporaryCorpTypeCode')
+            if draft['draftType'] == 'ATMP' and draft['legalName'] is None:
+                draft['legalName'] = 'Numbered Amalgamated Company'
```

Comment:
Alternative to this, would be doing this:
https://github.com/bcgov/sbc-auth/pull/2705